pkg/config/config_test.go

```diff
@@ -540,13 +540,6 @@ func (s *MySuite) TestHasKind(c *C) {
 
 }
 
-func (s *MySuite) TestCheckModuleAndGroupNames(c *C) {
```

Comment:
The overall unit test coverage goes up by 0.1% but I think we are missing some important edge cases in this particular function.

After running `go tool cover -html=coverage.out hpc-toolkit/pkg/config` and `go tool cover -html=coverage.out` I see these fields:

```
                        if _, ok := seen[mod.ID]; ok {
                                return fmt.Errorf("%s: %s used more than once", errorMessages["duplicateID"], mod.ID)
                        }
```

and

```
                        } else if grp.Kind != mod.Kind {
                                return fmt.Errorf(
                                        "%s: deployment group %s, got: %s, wanted: %s",
                                        errorMessages["mixedModule"],
                                        grp.Name, grp.Kind, mod.Kind)
                        }
```

Reply:
This was probably true before, but I'd rather improve this along the way while we are here.

Reply:
Added test for ` used more than once `. But unearthed other problems in this function, fixing + adding test is going out of scope of this PR.